examples/controls_example/public/react_controls/control_panel.tsx

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import classNames from 'classnames';
+import React, { useState } from 'react';
+
+import { EuiFlexItem, EuiFormControlLayout, EuiFormLabel, EuiFormRow, EuiIcon } from '@elastic/eui';
+import { css } from '@emotion/react';
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+import { i18n } from '@kbn/i18n';
+import {
+  apiHasParentApi,
+  apiPublishesViewMode,
+  useBatchedOptionalPublishingSubjects,
+} from '@kbn/presentation-publishing';
+import { FloatingActions } from '@kbn/presentation-util-plugin/public';
+import { euiThemeVars } from '@kbn/ui-theme';
+
+import { ControlError } from './control_error_component';
+import { ControlPanelProps, DefaultControlApi } from './types';
+
+/**
+ * TODO: Handle dragging
+ */
+const DragHandle = ({ isEditable, controlTitle }: { isEditable: boolean; controlTitle?: string }) =>
+  isEditable ? (
+    <button
+      aria-label={i18n.translate('controls.controlGroup.ariaActions.moveControlButtonAction', {
+        defaultMessage: 'Move control {controlTitle}',
+        values: { controlTitle: controlTitle ?? '' },
+      })}
+      className="controlFrame__dragHandle"
+    >
+      <EuiIcon type="grabHorizontal" />
+    </button>
+  ) : null;
+
+export const ControlPanel = <ApiType extends DefaultControlApi = DefaultControlApi>({
+  Component,
+}: ControlPanelProps<ApiType>) => {
+  const [api, setApi] = useState<ApiType | null>(null);
+
+  const viewModeSubject = (() => {
```

Comment:
I took this idea from `presentation_panel_internal`, yeah - agreed that there is probably a better way of handling this, but going to leave it for now so that it can remain batched :)